### PR TITLE
feat: add acceptance tests

### PR DESCRIPTION
**Pull request by dimitarproynov**
**Summary of Pull Request**

Added acceptance tests for the following resources:
- CEIP
- Host
- Network Pool
- User

The acceptance tests are basic but will provide a level of certainty that the provider functions as intended. 

The provider and the resources had to be moved from package "vcf" to package "provider" because of Golang's restriction of import loops. 

Various Resource lifecycle functions have been fixed, so that the tests could pass.

**Type of Pull Request**

- [ ] This is a bug fix.
- [X] This is an enhancement or feature.
- [ ] This is a code style/formatting update.
- [x] This is a documentation update.
- [X] This is a refactoring update.
- [x] This is a chore update
- [ ] This is something else.
      Please describe:

**Related to Existing Issues**

Issue Number: N/A

**Test and Documentation Coverage**

For bug fixes or features:

- [X] Tests have been completed.
- [ ] Documentation has been added/updated.

Testing done:

```shell
make lint
make build
make test

make testacc
TF_ACC=1 go test $(go list ./... |grep -v 'vendor') -v  -timeout 240m -parallel=4
?   	github.com/vmware/terraform-provider-vcf	[no test files]
?   	github.com/vmware/terraform-provider-vcf/internal/constants	[no test files]
=== RUN   TestProvider
--- PASS: TestProvider (0.00s)
=== RUN   TestAccResourceVcfCeip
=== PAUSE TestAccResourceVcfCeip
=== RUN   TestAccResourceVcfHost
=== PAUSE TestAccResourceVcfHost
=== RUN   TestAccResourceVcfNetworkPool
=== PAUSE TestAccResourceVcfNetworkPool
=== RUN   TestAccResourceVcfUser
=== PAUSE TestAccResourceVcfUser
=== RUN   TestValidatePassword
=== RUN   TestValidatePassword/Validate_password
=== RUN   TestValidatePassword/Nil_password_validation
--- PASS: TestValidatePassword (0.00s)
    --- PASS: TestValidatePassword/Validate_password (0.00s)
    --- PASS: TestValidatePassword/Nil_password_validation (0.00s)
=== RUN   TestValidateParsingFloatToInt
--- PASS: TestValidateParsingFloatToInt (0.00s)
=== RUN   TestConvertToStringSlice
--- PASS: TestConvertToStringSlice (0.00s)
=== CONT  TestAccResourceVcfCeip
=== CONT  TestAccResourceVcfNetworkPool
=== CONT  TestAccResourceVcfHost
=== CONT  TestAccResourceVcfUser
--- PASS: TestAccResourceVcfNetworkPool (7.85s)
--- PASS: TestAccResourceVcfUser (10.46s)
--- PASS: TestAccResourceVcfCeip (50.40s)
--- PASS: TestAccResourceVcfHost (113.62s)
PASS
ok  	github.com/vmware/terraform-provider-vcf/internal/provider	114.079s
```

**Breaking Changes?**

- [ ] Yes, there are breaking changes.
- [X] No, there are no breaking changes.

### CODE REVIEW

**Comment by tenthirtyam** [comment]
thanks! i'll review this tomorrow.

File: examples/resources/host/variables.tf [hunk 1]
```diff
@@ -0,0 +1,29 @@
+variable "sddc_manager_username" {
+  description = "username used to authenticate against an SDDC Manager instance"
```
Comment (tenthirtyam) [inline examples/resources/host/variables.tf:2]:
```suggestion
  description = "Username used to authenticate against an SDDC Manager instance"
```

File: examples/resources/network_pool/variables.tf [hunk 2]
```diff
@@ -0,0 +1,14 @@
+variable "sddc_manager_username" {
+  description = "username used to authenticate against an SDDC Manager instance"
```
Comment (tenthirtyam) [inline examples/resources/network_pool/variables.tf:2]:
```suggestion
  description = "Username used to authenticate against an SDDC Manager instance"
```

File: examples/resources/user/variables.tf [hunk 3]
```diff
@@ -0,0 +1,29 @@
+variable "sddc_manager_username" {
+  description = "username used to authenticate against an SDDC Manager instance"
```
Comment (tenthirtyam) [inline examples/resources/user/variables.tf:2]:
```suggestion
  description = "Username used to authenticate against an SDDC Manager instance"
```

File: examples/resources/user/variables.tf [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+}
+
+variable "sddc_manager_host" {
+  description = "FQDN of an SDDC Manager instance"
+  default = ""
+}
+
+variable "sso_domain" {
+  description = "The SSO domain in which an SSO user is to be created"
+  default = "vrack.vsphere.local"
```
Comment (tenthirtyam) [inline examples/resources/user/variables.tf:18]:
```suggestion
  default = "vsphere.local"
```

File: go.mod [hunk 5]
```diff
@@ -3,12 +3,11 @@ module github.com/vmware/terraform-provider-vcf
 go 1.19
 
 require (
-	github.com/davecgh/go-spew v1.1.1
 	github.com/go-openapi/runtime v0.26.0
 	github.com/go-openapi/strfmt v0.21.7
 	github.com/hashicorp/terraform-plugin-docs v0.14.1
 	github.com/hashicorp/terraform-plugin-sdk/v2 v2.26.1
-	github.com/vmware/vcf-sdk-go v0.1.0
+	github.com/vmware/vcf-sdk-go v0.1.1
```
Comment (tenthirtyam) [inline go.mod:10]: There is an open PR for this. See #17

**Review by tenthirtyam — CHANGES_REQUESTED** [review]
Just a minor nit for 'username'.

File: go.mod (hunk 5 above)
Comment (dimitarproynov) [reply go.mod:10]: A manual intervention is required since the method names, provided by the SDK have changed "GETHTTP... -> GetHTTP...".

**Review by nklinkachevAtVMware — APPROVED** [review]

**Review by tenthirtyam — APPROVED** [review]

**Pull request merged**